### ISSUE TITLE
sycl_cts/test_range fails on win & L0 gen12

### ISSUE THREAD

**Issue by KornevNikita** [intel/llvm#19677]
### Describe the bug

See: https://github.com/intel/llvm/actions/runs/16665652646/job/47173694163
```
  -------------------------------------------------------------------------------
  range_api
  -------------------------------------------------------------------------------
  D:\github\_work\llvm\llvm\khronos_sycl_cts\tests\range\../common\../../util/proxy.h(47)
  ...............................................................................
  
  D:\github\_work\llvm\llvm\khronos_sycl_cts\util\sycl_exceptions.h(184): 
  warning:
    SYCL exception
    with category name: 'sycl'
    with code: 'build'
    with code raw value: '7'
    with code message: 'SYCL Error'
    with what: 'The program was built for 1 devices
    Build program log for 'Intel(R) Iris(R) Xe Graphics':
    IGC: Internal Compiler Error: Stack overflow
    '
  
  D:\github\_work\llvm\llvm\khronos_sycl_cts\util\test_base.cpp(28): FAILED:
  explicitly with message:
    a SYCL exception was caught: The program was built for 1 devices
    Build program log for 'Intel(R) Iris(R) Xe Graphics':
    IGC: Internal Compiler Error: Stack overflow
```
Most likely this fails after sycl-cts [patch](https://github.com/KhronosGroup/SYCL-CTS/pull/1128) by @HPS-1 - could you please take a look?

### To reproduce

1. Include a code snippet that is as short as possible
2. Specify the command which should be used to compile the program
3. Specify the command which should be used to launch the program
4. Indicate what is wrong and what was expected

### Environment

- OS: [e.g Windows/Linux]
- Target device and vendor: [e.g. Intel GPU]
- DPC++ version: [e.g. commit hash or output of `clang++ --version`]
- Dependencies version: [e.g. the output of `sycl-ls --verbose`]

**Comment by aelovikov-intel** [comment]
> Most likely this fails after sycl-cts https://github.com/KhronosGroup/SYCL-CTS/pull/1128 by @HPS-1 - could you please take a look?

@HPS-1 , can you confirm you're looking into this? Is there a quick workaround available? If not, please temporarily disable the test to make Nightly green.

**Comment by HPS-1** [comment]
> > Most likely this fails after sycl-cts [KhronosGroup/SYCL-CTS#1128](https://github.com/KhronosGroup/SYCL-CTS/pull/1128) by [@HPS-1](https://github.com/HPS-1) - could you please take a look?
> 
> [@HPS-1](https://github.com/HPS-1) , can you confirm you're looking into this? Is there a quick workaround available? If not, please temporarily disable the test to make Nightly green.

Yes, I'm currently looking into this. Given it's a stack overflow error and the large number of test cases added in that patch (and AFAIK Windows has a relatively small default stack size), my current guess is the reason is simply that there're too many test cases. I'm trying to reproduce the error and see if removing a few test cases will fix it. With that said, may I ask if the failing CTS tests are using code directly from the KhronosGroup/SYCL-CTS repo or some fork of it? (As I need to know where to push my fix to)

**Comment by KornevNikita** [comment]
@HPS-1 here in intel/llvm we use the [original sycl-cts repo](https://github.com/KhronosGroup/SYCL-CTS)

**Comment by KornevNikita** [comment]
AFAIU https://github.com/KhronosGroup/SYCL-CTS/pull/1140 should fix this.
